Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java

```diff
@@ -554,6 +558,28 @@ private void removeRepoFromCache(String authority) {
             if (notEmpty.contains(uri)) continue;
             deleteSeries(context, uri);
         }
+        
+        if (recurse) {
```

Comment:
I agree that the code looks identical, but deep inside the loop it calls listSCHEMEByUriPrefix which makes it difficult to handle generically (believe me, I already tried). 

If we changed the name of listSCHEMEByUriPrefix to listByUriPrefix then we could do it but the naming would be inconsistent (though I'd be prepared to support a motion of removing SCHEME from function names)

We could maybe handle this using the new Java 8 lambda expressions if that's an option.

I'm open to suggestions.